nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -383,11 +383,17 @@ impl ShardWriter {
         };
 
         let mut vector_tasks = vec![];
-        for (_, vector_writer) in indexes.vectors_indexes.iter_mut() {
+        for (vectorset, vector_writer) in indexes.vectors_indexes.iter_mut() {
             vector_tasks.push(|| {
                 run_with_telemetry(info_span!(parent: &span, "vector set_resource"), || {
                     debug!("Vector service starts set_resource");
-                    let result = vector_writer.set_resource(&resource);
+                    let vectorset_resource = match vectorset.as_str() {
+                        "" | DEFAULT_VECTORS_INDEX_NAME => nucliadb_core::vectors::ResourceWrapper::from(&resource),
```

Comment:
Usually we use `.into()` to call the `From` implementation. Also applies to some tests below

```suggestion
                        "" | DEFAULT_VECTORS_INDEX_NAME => (&resource).into(),,
```